DQMServices/Core/src/DQMStore.cc

```diff
@@ -99,15 +99,30 @@ namespace dqm::implementation {
       MonitorElementData medata;
       medata.key_.path_ = path;
       medata.key_.kind_ = kind;
-      medata.key_.scope_ = this->scope_;
+
+      bool pathInList = false;
+      if (not store_->MEsToSave_.empty()) {
+        for (std::vector<std::string>::const_iterator ipath = store_->MEsToSave_.begin();
+             ipath != store_->MEsToSave_.end();
+             ++ipath) {
+          if (fullpath == (*ipath)) {
```

Comment:
```suggestion
        for (auto& thepath : store_->MEsToSave_) {
          if (fullpath == thepath) {
```
(range based loop may work here)